src/act_wiz.cpp

```diff
@@ -3465,41 +3454,36 @@ void do_rset(CHAR_DATA *ch, const char *argument) {
 }
 
 void do_sockets(CHAR_DATA *ch, const char *argument) {
-    char buf[2 * MAX_STRING_LENGTH];
-    char buf2[MAX_STRING_LENGTH];
+    std::string buf;
     char arg[MAX_INPUT_LENGTH];
-    Descriptor *d;
-    int count;
-
-    count = 0;
-    buf[0] = '\0';
+    int count = 0;
 
     one_argument(argument, arg);
-    for (d = descriptor_list; d != nullptr; d = d->next) {
-        if (d->character() != nullptr && can_see(ch, d->character())
-            && (arg[0] == '\0' || is_name(arg, d->character()->name)
-                || (d->original() && is_name(arg, d->original()->name)))) {
-            count++;
-            bug_snprintf(buf + strlen(buf), sizeof(buf), "[%3d %5s] %s@%s\n\r", d->channel(), short_name_of(d->state()),
-                         d->person() ? d->person()->name : "(none)", d->host().c_str());
-        } else if (!d->character() && get_trust(ch) == MAX_LEVEL) {
-            /*
-             * New: log even connections that haven't logged in yet
-             * Level 100s only, mind
-             */
+    const auto view_all = arg[0] == '\0';
+    for (auto &d : descriptors().all()) {
+        const char *name = nullptr;
+        if (auto *victim = d.character()) {
+            if (!can_see(ch, victim))
+                continue;
+            if (view_all || is_name(arg, d.character()->name) || is_name(arg, d.person()->name))
+                name = d.person()->name;
+        } else if (get_trust(ch) == MAX_LEVEL) {
+            // log even connections that haven't logged in yet
+            // Level 100s only, mind
+            name = "(unknown)";
+        }
+        if (name) {
             count++;
-            bug_snprintf(buf + strlen(buf), sizeof(buf), "[%3d %5s] (unknown)@%s\n\r", d->channel(),
-                         short_name_of(d->state()), d->host().c_str());
+            buf += "[{:3} {:>5}] {}@{}\n\r"_format(d.channel(), short_name_of(d.state()), name, d.host().c_str());
```

Comment:
do you need the c_str() here?

Reply:
No!!! yay! Thanks! I have a mental TODO to look for c_str() cases that came along during integration work...thanks for eagle-eyed spotting that!